Rollbar/CustomKeyValuePackageDecorator.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace Rollbar
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Rollbar.DTOs;
+
+    /// <summary>
+    /// Class CustomKeyValuePackageDecorator.
+    /// Implements the <see cref="Rollbar.RollbarPackageDecoratorBase" />
+    /// </summary>
+    /// <seealso cref="Rollbar.RollbarPackageDecoratorBase" />
+    public class CustomKeyValuePackageDecorator
+        : RollbarPackageDecoratorBase
+    {
+        /// <summary>
+        /// The custom
+        /// </summary>
+        private readonly IDictionary<string, object> _custom;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CustomKeyValuePackageDecorator"/> class.
+        /// </summary>
+        /// <param name="packageToDecorate">The package to decorate.</param>
+        /// <param name="custom">The custom.</param>
+        public CustomKeyValuePackageDecorator(
+            IRollbarPackage packageToDecorate, 
+            IDictionary<string, object> custom
+            )
+            : this(packageToDecorate, custom, false)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CustomKeyValuePackageDecorator"/> class.
+        /// </summary>
+        /// <param name="packageToDecorate">The package to decorate.</param>
+        /// <param name="custom">The custom.</param>
+        /// <param name="mustApplySynchronously">if set to <c>true</c> [must apply synchronously].</param>
+        public CustomKeyValuePackageDecorator(
+            IRollbarPackage packageToDecorate, 
+            IDictionary<string, object> custom, 
+            bool mustApplySynchronously
+            ) 
+            : base(packageToDecorate, mustApplySynchronously)
+        {
+            this._custom = custom;
+        }
+
+        /// <summary>
+        /// Decorates the specified rollbar data.
+        /// </summary>
+        /// <param name="rollbarData">The rollbar data.</param>
+        protected override void Decorate(Data rollbarData)
+        {
+            if (this._custom == null || this._custom.Count == 0)
+            {
+                return; // nothing to decorate with...
+            }
+
+            if (rollbarData.Custom != null || rollbarData.Custom.Count == 0)
```

Comment:
Yeah I agree with codacy-bot here, this condition doesn't make any sense.

Reply:
actually, this one is correct the way it is:
if Custom is not initialized or initialized but empty - just assign new incoming custom collection,
otherwise, add the incoming collection to the existing (non-empty) one...

Reply:
"if Custom is not initialized" means `rollbarData.Custom == null` which is the opposite of what is here